acceptancetests/assess_lxd_space.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+"""Testing spaces and subnets settings for app deployment to lxd container.
+   This is an validation of Bug 1685782:
+   https://bugs.launchpad.net/juju/+bug/1685782
+   The test procedures are based on comment #29.
+"""
+
+from __future__ import print_function
+
+import os
+import sys
+import json
+import logging
+import argparse
+
+from jujucharm import local_charm_path
+from deploy_stack import BootstrapManager
+
+from utility import (
+    JujuAssertionError,
+    add_basic_testing_arguments,
+    configure_logging,
+    )
+
+__metaclass__ = type
+log = logging.getLogger("assess_lxd_space")
+
+
+def assert_initial_spaces(client):
+    """Initial spaces status after bootstrap should show as:
+       no spaces to display"""
+    # Run juju spaces --format json
+    # Bug 1704105: https://bugs.launchpad.net/juju/+bug/1704105
+    # merge_stderr need to be set to True for the initial assert
+    # because spaces list is empty at this point.
+    raw_output = client.get_juju_output(
+        'spaces', '--format', 'json', include_e=False, merge_stderr=True)
+    expected_spaces = 'no spaces to display'
+    if expected_spaces not in raw_output:
+        raise JujuAssertionError(
+            'Incorrect initial spaces status. Found: {}\nExpected: {}'.format(
+            raw_output, expected_spaces))
+    else:
+        log.info('Initial spaces status is: {}.'.format(expected_spaces))
+
+
+def get_subnets(client):
+    # Run juju subnets --format json
```

Comment:
But the command string you've commented is on the next line almost verbatim.
The comment also doesn't include the details like any environment variables that are set.
These details are shown i the log of a test run (with --debug set each command outputs the env-var dict but at normal logging all commands are logged out, arguments and all).

There is also a log file (actually a yaml file) produce by a script run that outlines each juju command run (in order, includes datestamp etc.) so one can see the whole history of juju commands run during a test run. Admittedly this is after a test run, and not just reading the source of the test.

I'll concede that perhaps I'm too familar with the testing framework to see the value in such comments, so it would be useful to get the input of someone that needs to look at tests infrequently (not necessarily for this specific point, but details like this in general.)